### ISSUE TITLE
 warning: unknown ZLLSwitch sensor  error

### ISSUE THREAD

**Issue by pponce** [ebaauw/homebridge-hue#108]
I recently started seeing these warnings and am not sure what they are. This is for my V2 hue hub.

[5/3/2017, 5:18:43 PM] [Philips Hue 1] Philips hue: /sensors/6: warning: unknown ZLLSwitch sensor {"state":{"buttonevent":1002,"lastupdated":"2017-05-03T15:15:41"},"config":{"on":true,"battery":100,"reachable":true,"pending":[]},"name":"Master Bedroom dimmer","type":"ZLLSwitch","modelid":"RWL020","manufacturername":"Philips","swversion":"5.45.1.17846","uniqueid":"xx:xx:xx:xx:xx:xx:xx:xx-xx-xxxx"}
[5/3/2017, 5:18:43 PM] [Philips Hue 1] Philips hue: /sensors/8: warning: unknown ZLLSwitch sensor {"state":{"buttonevent":4003,"lastupdated":"2017-05-03T15:54:55"},"config":{"on":true,"battery":100,"reachable":true,"pending":[]},"name":"TV Room dimmer","type":"ZLLSwitch","modelid":"RWL020","manufacturername":"Philips","swversion":"5.45.1.17846","uniqueid":"xx:xx:xx:xx:xx:xx:xx:xx-xx-xxxx"}
[5/3/2017, 5:18:43 PM] [Philips Hue 1] Philips hue: /sensors/11: warning: unknown ZLLSwitch sensor {"state":{"buttonevent":4002,"lastupdated":"2017-05-02T03:48:49"},"config":{"on":true,"battery":null,"reachable":false,"pending":[]},"name":"Office dimmer","type":"ZLLSwitch","modelid":"RWL020","manufacturername":"Philips","swversion":"5.45.1.16265","uniqueid":"xx:xx:xx:xx:xx:xx:xx:xx-xx-xxxx"}
[5/3/2017, 5:18:43 PM] [Philips Hue 1] Philips hue: /sensors/13: warning: unknown ZLLSwitch sensor {"state":{"buttonevent":4003,"lastupdated":"2017-05-03T04:41:06"},"config":{"on":true,"battery":100,"reachable":true,"pending":[]},"name":"Stairs dimmer bottom","type":"ZLLSwitch","modelid":"RWL020","manufacturername":"Philips","swversion":"5.45.1.16265","uniqueid":"xx:xx:xx:xx:xx:xx:xx:xx-xx-xxxx"}
[5/3/2017, 5:18:43 PM] [Philips Hue 1] Philips hue: /sensors/15: warning: unknown ZLLSwitch sensor {"state":{"buttonevent":null,"lastupdated":"none"},"config":{"on":true,"battery":100,"reachable":true,"pending":[]},"name":"Stairs dimmer top","type":"ZLLSwitch","modelid":"RWL020","manufacturername":"Philips","swversion":"5.45.1.17846","uniqueid":"xx:xx:xx:xx:xx:xx:xx:xx-xx-xxxx"}
[5/3/2017, 5:18:43 PM] [Philips Hue 1] Philips hue: /sensors/17: warning: unknown ZLLSwitch sensor {"state":{"buttonevent":1002,"lastupdated":"2017-05-03T01:19:09"},"config":{"on":true,"battery":100,"reachable":true,"pending":[]},"name":"Dining room dimmer","type":"ZLLSwitch","modelid":"RWL020","manufacturername":"Philips","swversion":"5.45.1.17846","uniqueid":"xx:xx:xx:xx:xx:xx:xx:xx-xx-xxxx"}
[5/3/2017, 5:18:43 PM] [Philips Hue 1] Philips hue: /sensors/21: warning: unknown ZLLSwitch sensor {"state":{"buttonevent":4003,"lastupdated":"2017-05-02T04:02:06"},"config":{"on":true,"battery":100,"reachable":true,"pending":[]},"name":"Living Room dimmer","type":"ZLLSwitch","modelid":"RWL020","manufacturername":"Philips","swversion":"5.45.1.17846","uniqueid":"xx:xx:xx:xx:xx:xx:xx:xx-xx-xxxx"}
[5/3/2017, 5:18:43 PM] [Philips Hue 1] Philips hue: /sensors/35: warning: unknown ZLLSwitch sensor {"state":{"buttonevent":null,"lastupdated":"none"},"config":{"on":true,"battery":100,"reachable":true,"pending":[]},"name":"Hallway dimmer","type":"ZLLSwitch","modelid":"RWL020","manufacturername":"Philips","swversion":"5.45.1.17846","uniqueid":"xx:xx:xx:xx:xx:xx:xx:xx-xx-xxxx"}

**Comment by ebaauw** [comment]
I started whitelisting known sensors, in my efforts to support deCONZ and the Ikea Trådfri remote (see issue #106) .  I didn't know the the Hue dimmer switches also use `modelid` RWL020 - mine show as RWL021.  I'll add them.

**Comment by ebaauw** [comment]
Added RWL020 to v0.4.2.

Interesting: your Hue Dimmer switches report the same two values for `swversion` as mine.  The ones with 5.45.1.16265 have a different regulatory label (US vs EU?) printed on the back, but the same model: 324131137411.  I recently got a new one, also modelid RWL021, same printed model, yet another regulatory label (Chinese?), but also swversion 5.45.1.17846.

The Hue dimmer switches with the older swversion expose themselves as (slightly) different devices on the ZigBee network (see https://github.com/ebaauw/homebridge-hue/wiki/ZigBee-Devices#hue-dimmer-switch), but from a Hue bridge perspective, they have the same functionality.  I don't know why the Hue bridge hasn't updated their firmware.
